datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dashboard/mappers/DashboardMapper.java

```diff
@@ -79,6 +81,14 @@ public Dashboard apply(@Nonnull final EntityResponse entityResponse) {
                 result.setInstitutionalMemory(InstitutionalMemoryMapper.map(new InstitutionalMemory(data)));
             } else if (GLOSSARY_TERMS_ASPECT_NAME.equals(name)) {
                 result.setGlossaryTerms(GlossaryTermsMapper.map(new GlossaryTerms(data)));
+            } else if (DOMAINS_ASPECT_NAME.equals(name)) {
+                final Domains domains = new Domains(data);
+                // Currently we only take the first domain if it exists.
+                if (domains.getDomains().size() > 0) {
+                    result.setDomain(Domain.builder()
+                        .setType(EntityType.DOMAIN)
+                        .setUrn(domains.getDomains().get(0).toString()).build());
+                }
```

Comment:
shouldn't we not need to do this bc the domains aspect isn't in the snapshot?

Reply:
We are not using snapshots anymore! We're using batchGetV2, which fetches generic aspects. We still fetch in batch, as opposed to having 1 call per aspect